### ISSUE TITLE
Accessibility issues

### ISSUE THREAD

**Issue by zcorpan** [pcottle/learnGitBranching#960]
Hi!

I tried using macOS VoiceOver to navigate https://learngitbranching.js.org/ and noticed some accessibility issues:

- The "Select a level" dialog is always present as seen by the screen reader (I guess it's positioned off-screen, should be `display: none` or so when not visible)
- Level links are not links.
- Next/previous buttons are not buttons (not focusable and no text, so impossible to reach with VoiceOver).
- The `textarea` doesn't have a label.
- Content outside modal dialog boxes are not inert and so can be reached with screen readers. `<dialog>.showModal()` could help (or maybe the `inert` attribute).
- No landmarks (e.g. `helperBarMount` could be a `<nav>`).
- The visual representation as SVG is navigable with VoiceOver and it can access the text, but it's probably still not understandable for blind users. This one might be tricky to solve, though!

For people with low vision or color vision deficiency:

- Some text have poor color contrast, e.g. after solving a level, the "Confirm" and "Cancel" buttons.
- Some text is small, e.g. "`You can hide this window with "hide goal"`".

**Comment by ulyssear** [comment]
Some PRs have been created with some points solved (i guess) :

>  Content outside modal dialog boxes are not inert and so can be reached with screen readers. <dialog>.showModal() could help (or maybe the inert attribute).

Done with #1022 

>  After typing a command (e.g. git status), the resulting text that appears in the terminal should be read (it can be a live region, e.g. <output>). Compare with how VoiceOver works in the macOS Terminal app.
>  After typing a command, if the graph changes (e.g. git commit), the change should be announced (with a live region, or maybe as additional text in the terminal output).

Done with #1023. I haven't tried VoiceOver (or equivalent) yet.

**Comment by waldyrious** [comment]
> Done with #1023.

For reference, this was reverted e11d691480 in following the report in #1026.

**Comment by ulyssear** [comment]
> > Done with #1023.
> 
> For reference, this was reverted [e11d691](https://github.com/pcottle/learnGitBranching/commit/e11d69148086411b4ab3e030512d569d0a262a1b) in following the report in #1026. The approach may need to be refined.

Since the issue has been solved, do we put the <output> elements back or use the aria attributes instead?

**Comment by pcottle** [comment]
I think aria labels would be a lot safer in terms of browser support

**Comment by ulyssear** [comment]
>  Level links are not links.

Done with #1030

**Comment by shushiro** [comment]
Is anyone working on improvements "for people with low vision or color vision deficiency"? If not, I would like to take care of that :)

**Comment by pcottle** [comment]
I'm actually colorblind myself! I'm only the most common colorblind (red/green) so I haven't had any major issues using the application. Color isn't terribly important in our case, it's mostly just a visual aid to show which branches correspond to which commits.

I think the other accessibility issues above are more important / pressing.

**Comment by shushiro** [comment]
Alright! I then would love to have a go on one of them! How about the " Level links are not links." I would like to try keeping it small since it is my first real contribution to an open source project.
